src/System.Data.SqlClient/src/System/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -1420,20 +1420,27 @@ private object ConvertValue(object value, _SqlMetaData metadata, bool isNull, re
                             sqlValue = new SqlDecimal((Decimal)value);
                         }
 
+                        SqlDecimal sqlValueBeforeAdjust = sqlValue;
```

Comment:
Right, I knew it. But, in my understanding, the error message should show customer's input, not a value in the middle of process.